### PR TITLE
AMP-114937 update taxonomy api docs to support new delete/restore functionality

### PR DESCRIPTION
**Pull request by rahulmuthyam**
Taxonomy API doc additions and updates for Delete and Restore ops for Events, Event Properties and User Properties.

### CODE REVIEW

File: content/collections/api/en/taxonomy.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ eu_endpoint: 'https://analytics.eu.amplitude.com/api/2/'
 postman_link: 'https://www.postman.com/amplitude-dev-docs/workspace/amplitude-developers/folder/20044411-4b8180c4-e06e-478a-9c96-0ccf6cf652e2?action=share&source=copy-link&creator=29131806&ctx=documentation'
 api_status: ga
 lede: |-
-  The Taxonomy API grants Enterprise users the ability to programmatically plan their event schema in the Taxonomy tab.
+  The Taxonomy API grants `Enterprise` users the ability to programmatically plan their event schema in the Taxonomy tab.
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:12]:
```suggestion
  The Taxonomy API grants Enterprise users the ability to programmatically plan their event schema in the Taxonomy tab.
```

No need to add code formatting to the word Enterprise

File: content/collections/api/en/taxonomy.md [hunk 2]
```diff
@@ -810,7 +809,15 @@ Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA=
 
 |<div class="big-column">Name</div>| Description|
 |----|-----|
-|`event_type`| <span class="required">Required</span>. The name of the event type. Prefix custom event types with `ce:`. |
+|`event_type`| <span class="required">Required</span>. The name of the event type.|
+
+#### Behavior
+
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:815]: Let's make this list a two column table

File: content/collections/api/en/taxonomy.md [hunk 3]
```diff
@@ -822,16 +829,96 @@ A successful request returns a `200 OK` status and a JSON body.
 }
 ```
 
-#### 409 conflict response
+#### 4XX responses
 
-If there is a problem with your request, the request returns a `409 Conflict` status, and a JSON body with more information.
+If there is a problem with your request, the request returns a `4XX` status, and a JSON body with more information.
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:834]:
```suggestion
If there's a problem with the request, Amplitude returns a `4XX` status, and a JSON body with more information.
```

File: content/collections/api/en/taxonomy.md [hunk 4]
```diff
@@ -822,16 +829,96 @@ A successful request returns a `200 OK` status and a JSON body.
...
+
+```json
+{
+    "success": true
+}
+```
+
+#### 4XX responses
+
+If there is a problem with your request, the request returns a `4XX` status, and a JSON body with more information.
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:914]:
```suggestion
If there's a problem with your request, Amplitude returns a `4XX` status, and a JSON body with more information.
```

File: content/collections/api/en/taxonomy.md [hunk 5]
```diff
@@ -1309,8 +1396,83 @@ event_type=Onboarding%20Start
 
 |<div class="big-column">Name</div>|Description|
 |-----|---------|
-|`event_type`|<span class="required">Required</span>. Name of the event type to which the event properties belong to.|
+|`event_type`|<span class="optional">Optional</span>. Name of the event type to which the event properties belong to.|
+
+#### Behavior
+
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:1402]: Similar to above, this should be more readable as a table

File: content/collections/api/en/taxonomy.md [hunk 6]
```diff
@@ -1744,6 +1904,15 @@ Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA=
 |-----|---------|
 |`user_property`|<span class="required">Required</span>. The user property name. Prefix custom user properties with `gp:`.|
 
+#### Behavior
+
```
Comment (markzegarelli) [inline content/collections/api/en/taxonomy.md:1908]: Let's do a table here.

**Review by markzegarelli — CHANGES_REQUESTED** [review]
A few requests. I'll tackle the list -> table comments myself tomorrow.

File: content/collections/api/en/taxonomy.md [hunk 7]
```diff
@@ -603,20 +603,20 @@ Authorization: Basic {api-key}:{secret-key} #credentials must be base64 encoded
 {{/partial:tabs}}
 
 {{partial:collapse name="Example: Get an event type by name"}}
-This example gets the "Event 2" event type. This is a custom event, so it has a `ce:` prefix.
+This example gets the "Event 2" event type.
```
Comment (SpencerFleury) [inline content/collections/api/en/taxonomy.md:606]:
```suggestion
This example gets the 'Event 2' event type.
```

File: content/collections/api/en/taxonomy.md [hunk 8]
```diff
@@ -690,12 +690,12 @@ category=NEW_CATEGORY_NAME&display_name=NEW_EVENT_TYPE_DISPLAY_NAME
 {{/partial:tabs}}
 
 {{partial:collapse name="Example: Update an event type"}}
-This example updates the event type "OnboardingBegin" with the category "Onboarding", event type name "OnboardStart", the display name "Onboarding Start", and a description of "User signed in and completed an onboarding task from modal". Because the event type is custom, it has the `ce:` prefix.
+This example updates the event type "OnboardingBegin" with the category "Onboarding", event type name "OnboardStart", the display name "Onboarding Start", and a description of "User signed in and completed an onboarding task from modal".
```
Comment (SpencerFleury) [inline content/collections/api/en/taxonomy.md:693]:
```suggestion
This example updates the event type `OnboardingBegin` with the category `Onboarding`, event type name `OnboardStart`, the display name "Onboarding Start", and a description of "User signed in and completed an onboarding task from modal".
```

**Review by SpencerFleury — CHANGES_REQUESTED** [review]

**Review by markzegarelli — APPROVED** [review]

**Comment by rahulmuthyam** [comment]
Thanks for the review @markzegarelli and @SpencerFleury!

File: content/collections/api/en/taxonomy.md (hunk 1 above)
Comment (rahulmuthyam) [reply content/collections/api/en/taxonomy.md]: Some of our customers (who are not on an enterprise plan) kinda missed this part and raised some tickets related to this and I thought highlighting the plan name will at least reduce such instances.

**Review by SpencerFleury — APPROVED** [review]

**Pull request merged**